src/main/java/ui/TestController.java

```diff
@@ -79,4 +84,46 @@ public static boolean isCloseOnQuit() {
         return hasUI() && commandLineArgs.getOrDefault("closeonquit", "false").equalsIgnoreCase("true");
     }
 
+    /**
+     * Creates a Preferences that store data in test config file if run with test mode
+     * or in a default config file specified in the Preferences class
+     * @return
+     */
+    public static Preferences createApplicationPreference() {
+        if (isTestMode()) {
+            return createTestPreferences();
+        }
+
+        return new Preferences(Optional.empty());
+    }
+
+    /**
+     * Creates a Preferences that stores data in the test directory
+     * @return
+     */
+    public static Preferences createTestPreferences() {
+        return new Preferences(Optional.of(Preferences.TEST_CONFIG_FILE));
+    }
+
+    /**
+     * Creates a RepoIO for the application that uses different components
+     * depending on various test options: --test, --testjson etc.
+     * @return
+     */
+    public static RepoIO createApplicationRepoIO() {
+        return new RepoIO(isTestMode() ? Optional.of(new DummySource()) : Optional.empty(),
+                          isTestJSONEnabled() ? Optional.of(new JSONStoreStub()) : Optional.empty(),
+                          isTestMode() ? Optional.of(RepoStore.TEST_DIRECTORY) : Optional.empty());
+    }
+
+    /**
+     * Creates a partially stubbed RepoIO used for testing
+     * @param enableJSONStore set to true if actual json file is to be used for storing repos' data
+     * @return
+     */
+    public static RepoIO createTestingRepoIO(boolean enableJSONStore) {
```

Comment:
I think `createTestingRepoIO(false)` can be misleading. Does it mean we are creating an actual repo IO (the convention you used previously in this PR)? Would it be better to pass `Optional<JSONStore> jsonStoreToBeUsed` instead for the parameter? It can use `Optional.empty()` by default.